### PR TITLE
[14.0][MIG] purchase_request

### PR DESCRIPTION
**Pull request by HviorForgeFlow**
Standard migration to 14.0

- [x] Apply https://github.com/OCA/purchase-workflow/pull/999
- [x] Apply https://github.com/OCA/purchase-workflow/pull/996

CC @ForgeFlow

### CODE REVIEW

**Review by AaronHForgeFlow — COMMENTED** [review]
Functional tests looks great so far. Including mixing purchase requests lines, cancellations, notifications to requested by users and applying the minimum order quantity! Cool.

**Comment by HviorForgeFlow** [comment]
Weird behavior on invisible tag:
Using attrs:
image
Result:
image

Using "legacy" invisible, I see is still used in some views:
image

Result:
image

@dreispt

**Comment by JordiMForgeFlow** [comment]
Functional tests review.
It may be useful to have a string naming the checkbox for the product configuration.
image

**Comment by AaronHForgeFlow** [comment]
> Weird behavior on invisible tag:
> Using attrs:
> image
> Result:
> image
> 
> Using "legacy" invisible, I see is still used in some views:
> image
> 
> Result:
> image
> 
> @dreispt

Perhaps attrs requires a domain, or a condition?

**Comment by HviorForgeFlow** [comment]
I've tried using [(1, ,'=', 1)] domain and it's still showing :disappointed:

**Comment by HviorForgeFlow** [comment]
> Functional tests review.
> It may be useful to have a string naming the checkbox for the product configuration.
> image

solved!

**Comment by AaronHForgeFlow** [comment]
> I've tried using [(1, ,'=', 1)] domain and it's still showing

:confused:  better leave it as invisible="1" then

**Review by JordiMForgeFlow — APPROVED** [review]
Reviewed changes

File: purchase_request/models/purchase_request.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+        elif lines:
+            action["views"] = [
+                (self.env.ref("purchase.purchase_order_form").id, "form")
+            ]
+            action["res_id"] = lines.id
+        return action
+
+    @api.depends("line_ids")
+    def _compute_move_count(self):
+        self.move_count = len(
```
Comment (digitalsatori) [inline purchase_request/models/purchase_request.py]: self may not be a singleton, need a for loop here

File: purchase_request/models/purchase_request.py [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+    move_count = fields.Integer(
+        string="Stock Move count", compute="_compute_move_count", readonly=True
+    )
+    purchase_count = fields.Integer(
+        string="Purchases count", compute="_compute_purchase_count", readonly=True
+    )
+
+    @api.depends("line_ids")
+    def _compute_purchase_count(self):
+        self.purchase_count = len(self.mapped("line_ids.purchase_lines.order_id"))
```
Comment (digitalsatori) [inline purchase_request/models/purchase_request.py]: `self` may not be a singleton, need a for loop here.

**Review by digitalsatori — CHANGES_REQUESTED** [review]

**Comment by HviorForgeFlow** [comment]
@digitalsatori comments attended, thanks for your review.

**Review by AaronHForgeFlow — APPROVED** [review]

**Review by digitalsatori — APPROVED** [review]
LGTM

**Comment by HviorForgeFlow** [comment]
It is not ready yet, let me add remaining forward port

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
#996 is still pending

**Comment by HviorForgeFlow** [comment]
@AaronHForgeFlow company dependant feature added

**Review by Saran440 — APPROVED** [review]
Fuctional Test. LGTM :+1:

**Review by AaronHForgeFlow — APPROVED** [review]

**Comment by AaronHForgeFlow** [comment]
Nice!:
purchase_request

**Comment by youring** [comment]
https://github.com/ForgeFlow/purchase-workflow/pull/24
I just added a new icon for eye candy, pls merge. @HviorForgeFlow  Thanks.
Note:  for Odoo CE you have to install web_responsive to view icons https://github.com/OCA/web/pull/1711

**Review by AaronHForgeFlow — APPROVED** [review]

**Review by AaronHForgeFlow — APPROVED** [review]

**Comment by Saran440** [comment]
@HviorForgeFlow This PR is ready, Could you merged, please?

**Comment by levkar** [comment]
`track_visibility="onchange",`
should become
`tracking=True,`

occurs many times.

**Pull request merged**